src/app/plugins/connectfour.plugin.ts

```diff
@@ -0,0 +1,250 @@
+        // Only target our game emojis and no bot reactions
+        ConnectFourPlugin.moves.includes(react.emoji.name) && user.id != msg.author.id,
+      {
+        time: moment.duration(10, 'minutes').asMilliseconds(),
+      }
+    );
+    game.collector = collector;
+
+    collector.on('collect', async (react: MessageReaction) => {
+      // Last reaction.
+      const user = react.users.cache.last();
+      if (user !== game.getCurrentPlayer()) {
+        // Dump reactions from invalid players.
+        await react.users.remove(user);
+        return;
+      }
+
+      await game.move(ConnectFourPlugin.moves.indexOf(react.emoji.name), msg);
+
+      if (game.getWinner() || game.getTie()) {
+        collector.stop();
+        return;
+      }
+
+      await react.users.remove(user);
+    });
+
+    collector.on('end', () => {
+      msg.reactions.removeAll();
+    });
+  }
+}
+
+class ConnectFourGame {
+  public collector?: ReactionCollector;
+
+  private _playerA: User;
+  private _playerB: User;
+  private _board: number[][];
+  private _background: string = '🟦';
+  private _flagToEmoji: Record<number, string> = {
+    [-1]: '🔴',
+    [0]: '󠀠󠀠󠀠⬛',
+    [1]: '🟡',
+  };
+  private _rows: number = 6;
+  private _cols: number = 7;
+
+  private currentPlayer: number = -1;
+
+  private winner: Maybe<number> = null;
+  private tie: boolean = false;
+
+  private _searchDX: number[] = [-1, -1, 0, 1, 1, 1, 0, -1];
+  private _searchDY: number[] = [0, -1, -1, -1, 0, 1, 1, 1];
+
+  constructor(playerA: User, playerB: User) {
+    this._playerA = playerA;
+    this._playerB = playerB;
+
+    this._board = Array.from(Array(this._rows), (_) => Array(this._cols).fill(0));
+  }
+
+  public getCurrentPlayer() {
+    return this.currentPlayer === -1 ? this._playerA : this._playerB;
+  }
+
+  public getWinner() {
+    return this.winner;
+  }
+
+  public getTie() {
+    return this.tie;
+  }
+
+  public async move(col: number, msg: IMessage) {
+    // Make move.
+    const successfulDrop: boolean = this.dropPiece(col);
+    if (!successfulDrop) {
+      return;
+    }
+
+    // Check for win.
+    if (this.checkWin()) {
+      this.winner = this.currentPlayer;
+    } else {
+      this.tie = this.checkTie();
+
+      // Change players.
+      this.currentPlayer *= -1;
+    }
+
+    await msg.edit(this.showBoard());
+
+    return;
+  }
+
+  public dropPiece(col: number): boolean {
+    // Find first obstruction from top down in column.
+    let row = 0;
+    while (row < this._rows && this._board[row][col] === 0) {
+      row++;
+    }
+    // Subtract one to place piece one slot above obstruction.
+    row--;
+
+    if (row === -1) {
+      return false;
+    }
+
+    this._board[row][col] = this.currentPlayer;
+
+    return true;
+  }
+
+  public checkWin(): boolean {
+    // Returns if a postion is valid and owned by the current player.
+    const validAndOwned: (arg0: number, arg1: number) => boolean = (row, col) => {
```

Comment:
imo overkill to type functions, but if u feel strongly about it, maybe consider having more relevant var names for ur types instead of `arg0`/`arg1`

Reply:
Agreed! Hadn't written TypeScript before this project so it's typing schema is new to me. My editor suggested the [arg0, arg1, ...] so I thought it was forced. Making the change now :D